freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -98,72 +120,232 @@ def _get_requests_session(cls, retry_options={}):
-        errata_rpm_nvrs = errata.get_cve_affected_rpm_nvrs(event.advisory.errata_id)
+        errata_rpm_nvrs = self.errata.get_cve_affected_rpm_nvrs(
+            self.event.advisory.errata_id
+        )
         lb = LightBlue(
             server_url=conf.lightblue_server_url,
             cert=conf.lightblue_certificate,
             private_key=conf.lightblue_private_key,
             event_id=self.current_db_event_id,
         )
-        images = lb.get_images_by_nvrs(rebuild_images_list, rpm_nvrs=errata_rpm_nvrs)
+        images = lb.get_images_by_nvrs(images, rpm_nvrs=errata_rpm_nvrs)
         return images
+
+    def _outdated_composes(self, original_odcs_compose_ids, module_name_stream_set):
+        outdated_composes = set()
+        for compose_id in original_odcs_compose_ids:
+            compose = self.odcs.get_compose(compose_id)
+            source_type = compose.get("source_type")
+            if source_type != PungiSourceType.MODULE:
+                outdated_composes.add(compose_id)
+                continue
+
+            name_stream_set = set()
+            source_value = compose.get("source", "")
+            for source in source_value.split():
+                source = source.strip()
+                name_stream = source.rsplit(":", 2)[0]
+                name_stream_set.add(name_stream)
+
+            # If all the sources are not in advisory, renew them
+            inter_section = name_stream_set.intersection(module_name_stream_set)
+            if not inter_section:
+                outdated_composes.add(compose_id)
+
+        return outdated_composes
+
+    def _missing_composes(
+        self,
+        original_odcs_compose_ids,
+        module_name_stream_set,
+        module_name_stream_version_set,
+    ):
+        missing_composes = set()
+        for compose_id in original_odcs_compose_ids:
+            compose = self.odcs.get_compose(compose_id)
+            source_type = compose.get("source_type")
+            if source_type == PungiSourceType.MODULE:
+                name_stream_set = set()
+                mapping = dict()
+                source_value = compose.get("source", "")
+                for source in source_value.split():
+                    source = source.strip()
+                    name_stream = source.rsplit(":", 2)[0]
+                    name_stream_version = source.rsplit(":", 1)[0]
+                    name_stream_set.add(name_stream)
+                    mapping[name_stream] = name_stream_version
+
+                inter_section = name_stream_set.intersection(module_name_stream_set)
+                if inter_section:
+                    # If some module in advisory, create composes for all differences,
+                    # inter_section part will be added in following step
+                    difference = name_stream_set.difference(module_name_stream_set)
+                    for item in difference:
+                        missing_composes.add(mapping[item])
```

Comment:
```suggestion
                missing_composes.update(
                    mapping[name_stream]
                    for name_stream in name_stream_set.difference(module_name_stream_set)
                )
```

Reply:
I think we still need to check if the two sets are joint. I updated the codes, you can have a look with the new codes and see if it is proper.